stomp_core/src/stomp.cpp

```diff
@@ -788,16 +799,30 @@ bool Stomp::computeOptimizedCost()
     return false;
   }
 
+  // stop optimizing when valid solution is found
   if(current_lowest_cost_ > parameters_total_cost_)
   {
     current_lowest_cost_ = parameters_total_cost_;
   }
   else
   {
-    // reverting updates as no improvement was made
-    parameters_optimized_ -= parameters_updates_;
+    if(parameters_valid_)
+    {
+      if(parameters_valid_prev_)
+      {
+        // reverting updates as no improvement was made
+        parameters_optimized_ -= parameters_updates_;
+      }
+    }
+    else
+    {
+      // reverting updates as no improvement was made
+      parameters_optimized_ -= parameters_updates_;
```

Comment:
we need to set back parameters_valid_ here as well as the previous solution might have been valid:


```
      // is now valid again
      if (parameters_valid_prev_)
        parameters_valid_ = true;
```

or more simply

```
  // set back validity as well, previous might have been valid
 parameters_valid_ = parameters_valid_prev_;

```
this fixes "Case 1:"

Reply:
I've rearrange this part so as to set the current trajectory to valid when it was reverted to the previous valid one.